testmod/build.gradle

```diff
@@ -0,0 +1,139 @@
+    maven { url "https://jitpack.io" }
+    maven { url 'https://oss.sonatype.org/content/repositories/snapshots' }
+    mavenCentral()
+    jcenter()
+}
+
+dependencies {
+    // To change the versions see the gradle.properties file
+    minecraft "com.mojang:minecraft:${project.minecraft_version}"
+    mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
+    modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+
+    // Fabric API
+    modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
+
+    // Kotlin
+    modImplementation "net.fabricmc:fabric-language-kotlin:${project.fabric_kotlin_version}"
+    shade "org.jetbrains.kotlinx:kotlinx-serialization-core:${project.kx_ser_version}"
+    shade "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
+
+    // Debug
+    modRuntime "me.bymartrixx:wdmcf:1.0.2"
+
+    // PSA: Some older mods, compiled on Loom 0.2.1, might have outdated Maven POMs.
+    // You may need to force-disable transitiveness on them.
+}
+
+processResources {
+    inputs.property "version", project.version
+
+    filesMatching("fabric.mod.json") {
+        expand "version": project.version
+    }
+}
+
+// ensure that the encoding is set to UTF-8, no matter what the system default is
+// this fixes some edge cases with special characters not displaying correctly
+// see http://yodaconditions.net/blog/fix-for-java-file-encoding-problems-with-gradle.html
+tasks.withType(JavaCompile) {
+    options.encoding = "UTF-8"
+}
+
+// Loom will automatically attach sourcesJar to a RemapSourcesJar task and to the "build" task
+// if it is present.
+// If you remove this task, sources will not be generated.
+task sourcesJar(type: Jar, dependsOn: classes) {
+    classifier = "sources"
+    from sourceSets.main.allSource
+}
+
+jar {
+    from "LICENSE"
+}
+
+shadowJar {
+    from "LICENSE"
+
+    configurations = [ project.configurations.shade ]
+    classifier = "dev-all"
+    exclude "kotlin/**", "kotlinx/**", "javax/**", "META-INF"
+    exclude "org/checkerframework/**", "org/intellij/**", "org/jetbrains/annotations/**"
+    exclude "com/google/gson/**"
+    exclude "net/kyori/**"
+    exclude "org/slf4j/**"
+
+    def relocPath = "us.potatoboy.ledger.libs."
+    relocate "com.fasterxml", relocPath + "com.fasterxml"
+    relocate "com.moandjiezana.toml", relocPath + "com.moandjiezana.toml"
+    relocate "com.uchuhimo.konf", relocPath + "com.uchuhimo.konf"
+    relocate "javassist", relocPath + "javassist"
+    // Relocate each apache lib separately as just org.apache.commons will relocate things that aren't shaded and break stuff
+    relocate "org.apache.commons.lang3", relocPath + "org.apache.commons.lang3"
+    relocate "org.apache.commons.text", relocPath + "org.apache.commons.text"
+    relocate "org.reflections", relocPath + "org.reflections"
+    // it appears you cannot relocate sqlite due to the native libraries
+//    relocate "org.sqlite", relocPath + "org.sqlite"
+}
```

Comment:
Is there a reason all of the shadow stuff is repeated in the subproject buildscript, it doesn't really seem like its needed.

Reply:
As it turns out, the project builds (and runs in a production environment) without shadowing anything, so I just removed the shadow plugin.